### ISSUE TITLE
Documentation for modular Kyma

### ISSUE THREAD

**Issue by arturskorupa** [kyma-project/kyma#15437]
## Description

The concept of documenting modular Kyma is designed. The following scenarios (see Acceptance Criteria) should be taken into consideration (valid for OS and SKR documentation).

## Acceptance Criteria

- [x] Documentation (OS/SKR) of existing Kyma components has to be reviewed in terms of possible changes coming from the new modular achitecture. Discrepancies have to be adjusted.
- [x] Whenever possible, current structure of the documentation should be retained
- [x] Cross-cutting subjects (getting started guides, installation) have to be described in the docs (relates to https://github.com/kyma-project/community/issues/700)
- [x] Subjects specific to the new architecture (e.g module enable/disable) have to be described in the docs 
- [x] Module documentation should be decoupled from Kyma documentation (in case of deprecation/significant upgrades it would be easier to handle it)

**Links**
- [Concept document](https://github.com/kyma-project/community/tree/main/concepts/modularization)
- [Transition plan ](https://github.com/kyma-project/community/blob/main/concepts/modularization/transition.md)
- [Lifecycle manager](https://github.com/kyma-project/lifecycle-manager)
- [Module manager](https://github.com/kyma-project/module-manager)

**Comment by jacekon** [comment]
stacktrace of discussions on this subject:
_Hi, 
We discussed in the PO round what to do with the documentation in the modularization context. Couple of questions we got:
What to do with cross module documentation? Do we need scenario based guides at all? Maybe we should keep them independent? Do we need separate sections for operational guides, architecture & technical reference? Maybe simpler structure plus tagging? 
It was triggered by this thread: https://sap-btp.slack.com/archives/C01KSGZA23C/p1663143912856019
I wanted to have a quick chat with you about possible next steps, how to drive it and finally who should do it. 
I wanted to first get your opinion, before we start anything.

Cheers,
PB_

_Hi,
To sum up:
1.	The old navigation structure with modules listed on the left is preferred over the current layout.
2.	Each module should have (owns) its tutorial.
3.	We need to keep the ‘getting started’ section as an aggregate for the above tutorials (or a subset).
4.	Assumption: Components have their own separate repos; 
5.	Documentation for each component is in the respective component's repo; 
6.	Docs from all repos are consolidated on the Website, which serves as an entry point for the user.
7.	Tags as a complementary way of navigation
8.	Search is a must (it’s the key differentiator from GitHub)
9.	We need to involve Mariya for the UX proposals
10.	We need to involve Hasselhof/Varban as the current website owners for the implementation of above changes

Did I miss anything?
Regards,
Jacek_

**Comment by jacekon** [comment]
stacktrace of discussions on this subject:
_Hi PB,
1.	Are we going to create a strategy defining the relationship between OS and SKR docs, how we handle duplicated content vs. linking (and getting SKR readers stranded on the OS page…)?
2.	How is the 3-release-channel model planned to mesh with the continuous delivery setup for our SKR docs?
3.	How is the 3-release-channel model planned to mesh with the Unified Runtime (docu) strategy?
4.	Anything against reusing our old website structure? Of course, with adjustments to our current design.

Cheers,
Jacek_ 

_Thanks PB for the talk, to sum up for everyone:
1.	We continue with our current approach to have documentation from OS referenced/linked in SKR docs.
2.	There seem to be no constraint from the modularization at our docs, we will “simply” adopt the new approach from the SKR docs.
There will be two release channels: alpha and stable.
3.	UR docu strategy – no constraints from the modularization, here I will catch up with the respective UR docu people to sync our plans
4.	Nothing against adopting our old docu structure for the modules/components docus; As long as we have in parallel the task-based docu for the cross-cutting subjects e.g., getting started, installation, etc.

Thanks,
Jacek_

**Comment by jacekon** [comment]
- [ ] **Mariya** new UX proposal 
- [ ] @majakurcius module template based on old information/documentation types but simplified. With support from @NHingerl @grego952 @IwonaLanger. And taking into account requirements from the comments above (tutorial in each, common properties optional:true/false, etc.) 
- [ ] @valentinvieriu /Hasselhof: technical plan & implementation of the above

**Comment by jacekon** [comment]
stack trace of discussions on this project, from Maja:

_Hi! 

Below is my understanding of point #1 as how I understood also [@Bochynski, Piotr](mailto:piotr.bochynski@sap.com) would see it, and then some additional things we talked about.

Re 1:
I think if we want to go with the solution that I proposed on this oh so nicely drawn draft below (also attached) – i.e. the multiselect dropdown, from which you select individual components/a desired combination of components/ALL components (the first item on the list) – then:

1. The documentation can be browsed by components.
1a. The left-side navigation resembles what we have now, but…
1b. …the document categories are further simplified (i.e. Operation Guides and Developer Tutorials are merged into one category: Tutorials; The subcategories under Technical References could also be simplified and/or tidied up). 
This leaves us with only 4 main categories + Glossary: 
- Overview
- Get Started
- Tutorials
- Technical Reference 
Glossary
1c. This leaves us with a question: where do we move Troubleshooting and Security guides? Also to tutorials, under a dedicated sub-directory? Or would it be better to bring back separate main categories for these? [@DL Kyma TW](mailto:DL_5B9792637BCF8491C0000012@global.corp.sap) [@Bochynski, Piotr](mailto:piotr.bochynski@sap.com)
…
11. What about version on the Website? Tags and multiselect are fine if we only display the latest version. If we want to make it possible to select other versions (currently it’s 1/2 versions back or main, and pre-release), I imagine it either makes the version selection more complex (UX-wise? Probably something to consult with Mariya?), as it should be available to select for every component, and then also displayed for every component, or we have to give up multiselect? 
Unless we change the way we version the components – if we go with 2-3 [release channels](https://cloud.google.com/kubernetes-engine/docs/concepts/release-channels) (like stable/regular/rapid), and display docs for all components for the same channel, then that should not be an issue, we just have another dropdown for the channel version, collectively (or something of this sort). 
12. To ease up the transition from getting started with Kyma to further exploring it, we add Related links to each of the Get Started guide that would take the user to the relevant, connected documentation (just not too many, not to cause information overload; maybe to the related component’s overview or sth). 
13. Tutorials that touch more than one component are written in the main component’s repo (and own by the repo owners), but are also replicated/linked to in the other component’s repo, because if a user prefers to browse documentation via GitHub rather than the Website and/or is only interested in a particular component, they might miss it otherwise. 
13b. On the Website, such a document is nicely marked with the respective components’ tags (all the documents have tags, but such a tutorial with more “dependencies” is clearly spotted by having more than one component tag)._

**Comment by jacekon** [comment]
After a talk with Aneta Peteva, Nikola Simeonov, Astrid Herbst - we can't have more than one release channel on sap's help. Only one is supported by the current information structure in BTP docu area.

**Comment by majakurcius** [comment]
> After a talk with Aneta Peteva, Nikola Simeonov, Astrid Herbst - we can't have more than one release channel on sap's help. Only one is supported by the current information structure in BTP docu area.

Isn't this issue about the OS documentation on the kyma-project.io Website? It's also in the OS GitHub.

**Comment by NHingerl** [comment]
@jacekon  about "Module template based on SAP Help Portal and BTP information structure."

I'm still not sure what is meant with "Module template".

We're going to put explanatory content in [concept topics](https://help.sap.com/docs/DITA_CMS_User_Guide/bba968de69fc4814b97a0da987913be1/0119f5c7d085404a8867d88caa512ad4.html?state=DRAFT&locale=en-US), and instructional content in [task topics](https://help.sap.com/docs/DITA_CMS_User_Guide/bba968de69fc4814b97a0da987913be1/db23ac20be8447a1b8d3d44000634c8f.html?state=DRAFT&locale=en-US). The [DITA standard](https://help.sap.com/docs/DITA_CMS_User_Guide/bba968de69fc4814b97a0da987913be1/8f0c0efe7a524629b6d4f020c7d273e0.html?state=DRAFT&locale=en-US) has clear templates for those, no need to create them.

As for the information structure, we're bound to the BTP Core structure, which is sectioned with the following navigation nodes:

[Basic Platform Concepts](https://help.sap.com/docs/BTP/65de2977205c403bbc107264b8eccf4b/73beb06e127f4e47b849aa95344aabe1.html?locale=en-US)
[Getting Started](https://help.sap.com/docs/BTP/65de2977205c403bbc107264b8eccf4b/144e1733d0d64d58a7176e817fa6aeb3.html?locale=en-US)
[Development](https://help.sap.com/docs/BTP/65de2977205c403bbc107264b8eccf4b/c2fec62b49fa43b8bd945c85ecc2e5bd.html?locale=en-US)
[Extensions](https://help.sap.com/docs/BTP/65de2977205c403bbc107264b8eccf4b/08b1effc53634890a525f945017e2edc.html?locale=en-US)
[Administration and Operations](https://help.sap.com/docs/BTP/65de2977205c403bbc107264b8eccf4b/e183664210cf460796de3d90ca6bb6c3.html?locale=en-US)
[Security](https://help.sap.com/docs/BTP/65de2977205c403bbc107264b8eccf4b/e129aa20c78c4a9fb379b9803b02e5f6.html?locale=en-US)
[Getting Support](https://help.sap.com/docs/BTP/65de2977205c403bbc107264b8eccf4b/5dd739823b824b539eee47b7860a00be.html?locale=en-US)
[Glossary](https://help.sap.com/docs/BTP/65de2977205c403bbc107264b8eccf4b/e67a1435bb571014b2758553897c59b6.html?locale=en-US)


* Thus, we should put any explanatory overview information about modular Kyma in the introductory concept topic about the Kyma environment. I expect this to be just a few paragraphs; but if it's longer we can simply create a child topic below [Kyma Environment](https://help.sap.com/docs/BTP/65de2977205c403bbc107264b8eccf4b/468c2f3c3ca24c2c8497ef9f83154c44.html?locale=en-US).
  <img width="758" alt="image" src="https://user-images.githubusercontent.com/76950046/195571003-21055352-c135-4391-bb3c-31d8cedbb9b5.png">

* Information how to install and maintain modules is an Admin/Ops task, so that would become a new child topic under [Administration and Operations in the Kyma Environment](https://help.sap.com/docs/BTP/65de2977205c403bbc107264b8eccf4b/b8e16869e64a4abe93cc194aa6fdacf5.html?locale=en-US).
  <img width="331" alt="image" src="https://user-images.githubusercontent.com/76950046/195572035-c585ab90-6916-48f0-8513-f9074c5a27e8.png">



❓  I'm not sure what else we need? 
Looking at the email from Sept 6th, reviewing Kyma's overall content strategy for SKR docs is out of scope. 
➡️  <img width="544" alt="image" src="https://user-images.githubusercontent.com/76950046/195574003-0917abb5-d867-4757-beb9-47a0d5dfcceb.png">

So I propose that at the beginning of each existing "component/module" documentation (for example, [Kyma Metrics and Logs](https://help.sap.com/docs/BTP/65de2977205c403bbc107264b8eccf4b/d71a7cdb9c654860b41276b579d30da1.html?locale=en-US) or for Eventing, [Using Kyma Eventing with SAP Event Mesh](https://help.sap.com/docs/BTP/65de2977205c403bbc107264b8eccf4b/407d1266017f4b529b61665fa7408c41.html?locale=en-US)). we should simply link to the Admin/Ops topic that explains how to install a Kyma module. 

I don't know whether we have Kyma components/modules that don't have a Help Portal topic yet, but if those currently have no need for a separate topic, why would there be a need in the future?

**Comment by jacekon** [comment]
Hi @NHingerl @majakurcius I Nina's summary/example [from slack](https://sap-btp.slack.com/archives/C01KPEMJVGD/p1665410256943669) on how a module documentation could look and put it here: 
_from Nina Hingerl:
To illustrate what I mean, in practice that would mean the following, for example:
If the module team wants to explain the basic idea of their module and present the an overview, they’d pick the concept template.
For any step instructions, be it installing, configuring, any specific use cases, they’d pick the task template. (Edit: I hope installing the module works the same for all modules, so one common instructional doc should be sufficient, to which the module-specific documentation links as needed).
If there’s any other explanatory document needed to go into more detail than the overview provided, they pick the concept template and slot the doc into the navigation structure wherever it makes most sense.
If there’s more than one instruction needed (most likely yes), either sequentially or interspersed between other docs, they pick the task template again.
If they want to provide an overview introducing a sequence of several docs, they pick the concept template.
If there’s any typical troubleshooting, they pick the troubleshooting template and put it into the place where it fits best.
Imo, this should cover 80-90% of the required documentation. For any other docs, we could see if they fit into the aforementioned templates after some rephrasing, or whether they need another template._

**Comment by grego952** [comment]
[Here](https://sap-my.sharepoint.com/:w:/r/personal/grzegorz_karaluch_sap_com/_layouts/15/Doc.aspx?sourcedoc=%7BE46BA0D8-EE34-4DE8-B030-81C4547E6481%7D&file=mod_doc_concept_draft.docx&action=default&mobileredirect=true&isSPOFile=1&clickparams=eyJBcHBOYW1lIjoiVGVhbXMtRGVza3RvcCIsIkFwcFZlcnNpb24iOiIyOC8yMjExMzAwNDEwMCIsIkhhc0ZlZGVyYXRlZFVzZXIiOmZhbHNlfQ%3D%3D&cid=d6cf4db2-ed7e-49bb-9b99-9e94c3e55276) is a document that summarizes what we know so far.
Feel free to add any suggestions or questions.

**Comment by grego952** [comment]
The current documentation taks:
Prepare customer-facing documentation for Serverless Manager (https://github.com/kyma-project/serverless-manager/issues/33)
Prepare customer-facing documentation for Keda Manager (https://github.com/kyma-project/keda-manager/issues/80)
Prepare documentation common for all the modules (https://github.com/kyma-project/kyma/issues/16421)
Adjust the SKR documentation to the Modularization requirements (https://github.tools.sap/kyma/backlog/issues/3345)

**Comment by jacekon** [comment]
All acceptance criteria were discussed and ideas to address them were documented in the concept document linked [here](https://wiki.one.int.sap/wiki/display/kyma/Documentation+Concept)
It will take a while before each is implement, you can track progress [here](https://github.com/kyma-project/kyma/labels/docs%2Fmodularization) and [here](https://github.tools.sap/kyma/backlog/labels/docs%2Fmodularization):
